### PR TITLE
net: lib: nrf_cloud: send obj data for all topics

### PR DESCRIPTION
**Pull request by jayteemo**
Fixed an issue in nrf_cloud_send() that prevented data in the provided nrf_cloud_obj structure from being sent to the bulk and bin topics.
IRIS-9540

### CODE REVIEW

**Review by plskeggs — APPROVED** [review]

**Review by glarsennordic — APPROVED** [review]
I can't spot what actually changed, functionally -- this looks like a refactor to me with no functionality changes?

The refactor looks good, so I am approving, but can you point out the functionality difference? I apologize

**Comment by jayteemo** [comment]
> but can you point out the functionality difference? I apologize

for the bulk and bin topics:
```
.data.ptr = msg->data.ptr,
.data.len = msg->data.len,
```

should use the `send_data`

**Comment by glarsennordic** [comment]
Ahhh, thank you, I see it now

File: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -645,6 +645,8 @@ Libraries for networking
     * The JSON string representing longitude in ``PVT`` reports from ``lng`` to ``lon`` to align with nRF Cloud.
       nRF Cloud still accepts ``lng`` for backward compatibility.
 
+  * Fixed an issue in :c:func:`nrf_cloud_send` that prevented data in the provided :c:struct:`nrf_cloud_obj` structure from being sent to the bulk and bin topics.
```
Comment (wiba-nordic) [inline doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:648]:
```suggestion
  * Fixed an issue in the :c:func:`nrf_cloud_send` function that prevented data in the provided :c:struct:`nrf_cloud_obj` structure from being sent to the bulk and bin topics.
```

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

**Review by tony-le-24 — APPROVED** [review]
LGTM

**Review by wiba-nordic — APPROVED** [review]

**Pull request merged**